Assets/Scripts/Player/PlayerGroundCheck.cs

```diff
@@ -0,0 +1,21 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class PlayerGroundCheck : MonoBehaviour
+{
+    [SerializeField] private Transform m_groundCheckTF;
+    [SerializeField] private float m_groundCheckRadius = 0.2f;
+    [SerializeField] private LayerMask m_groundLayer;
+
+    public bool IsGrounded()
+    {
+        Collider[] hits = Physics.OverlapSphere(m_groundCheckTF.position, m_groundCheckRadius, m_groundLayer);
```

Comment:
This code causes a memory leak. Use https://docs.unity3d.com/ScriptReference/Physics.OverlapSphereNonAlloc.html